test/celeritas/decay/MuDecay.test.cc

```diff
@@ -0,0 +1,130 @@
+    auto const max_lepton_energy = real_type{0.5} * data_.muon_mass.value()
+                                   - data_.electron_mass.value();
+
+    // Anti-muon decay
+    {
+        this->set_inc_particle(pdg::mu_plus(), at_rest);
+        MuDecayInteractor interact(data_,
+                                   this->particle_track(),
+                                   this->direction(),
+                                   this->secondary_allocator());
+        auto result = interact(this->rng());
+
+        EXPECT_EQ(Interaction::Action::absorbed, result.action);
+        auto const& sec = result.secondaries;
+        EXPECT_EQ(1, sec.size());
+        EXPECT_EQ(pdg::positron(), params.id_to_pdg(sec[0].particle_id));
+        EXPECT_GE(max_lepton_energy, sec[0].energy.value());
+    }
+
+    // Muon decay
+    {
+        this->set_inc_particle(pdg::mu_minus(), at_rest);
+        MuDecayInteractor interact(data_,
+                                   this->particle_track(),
+                                   this->direction(),
+                                   this->secondary_allocator());
+        auto result = interact(this->rng());
+
+        EXPECT_EQ(Interaction::Action::absorbed, result.action);
+        auto const& sec = result.secondaries;
+        EXPECT_EQ(1, sec.size());
+        EXPECT_EQ(pdg::electron(), params.id_to_pdg(sec[0].particle_id));
+        EXPECT_GE(max_lepton_energy, sec[0].energy.value());
+    }
+}
+
+//---------------------------------------------------------------------------//
+TEST_F(MuDecayInteractorTest, stress_test)
+{
+    size_type const num_secondaries = 1;
+    size_type const num_samples = 10000;
+    MevEnergy const one_gev{1000};
+
+    this->resize_secondaries(num_secondaries * num_samples);
+    this->set_inc_particle(pdg::mu_minus(), one_gev);
+    MuDecayInteractor interact(data_,
+                               this->particle_track(),
+                               this->direction(),
+                               this->secondary_allocator());
+
+    Array<double, num_secondaries> avg_sec_energies{};  // Avg energy per sec
+    Real3 avg_total_momentum{};  // Avg total momentum per decay
+
+    for ([[maybe_unused]] auto sample : range(num_samples))
+    {
+        auto result = interact(this->rng());
+        auto const& sec = result.secondaries;
+
+        for (auto i : range(sec.size()))
+        {
+            auto const& part = sec[i];
+            avg_sec_energies[i] += part.energy.value();
+            axpy(part.energy.value(), part.direction, &avg_total_momentum);
+        }
+    }
+
+    avg_sec_energies /= num_samples;
+    avg_total_momentum /= num_samples;
+
+    // With only one secondary being returned, there is no expectation of
+    // energy or momentum conservation
+    static double const expected_avg_sec_energies[] = {384.834176348064};
+    static double const expected_avg_total_momentum[]
+        = {0.369721983532691, 0.109494401494642, 383.064043487629};
```

Comment:
Are these values consistent with Geant4 or your expectation?  It may be good to add a test with a low energy muon (e.g., 1 MeV).

Reply:
Thank you for adding a low energy muon test.

Actually, G4 has both `in-flight` and `at-rest`  muon decays.  Of course, the in-flight muon decay should compete with regular processes.  As the muon half lifetime is relatively long, a test detector with a large volume (order of 10 meters) is required, in which ionization will contribute to the final energy of muon before decays, but should be relatively minimal (MIP) for high energy muons with a sparse material (so that we may roughly assume that the total muon energy at the decay vertex is close to the input for GeV) - to be exact, universe or air may be the best material to test decay without loosing any enrgy.  No problem for leaving this test for later if it is difficult to isolate in-flight decay only as it will only check the boost effect for different input energies.  I do not think that we need to validate MC to experimental data (see for an example [here](https://www.researchgate.net/figure/Energy-spectrum-of-the-positron-in-Michel-decay-Points-data-Solid-histogram-Monte_fig3_237471229)) as the V-A calculation is a pure QED and should be (theoretically) good enough by itself (so, verification to Geant4 may be sufficient as Geant4 or other MC implementations are validated).